Karafka 2.0

mensfeld opened the pull request.

mensfeld commented:
It's pretty much done. I will be adding some small remarks down the road probably but would appreciate a review already.

mensfeld commented:
FYI:
a) I do not plan to merge it to master but to replace it
b) I squashed all of the changes

opti reviewed (approved):
Great work!

Any plans on adding 1.4 -> 2.0 upgrade guide? Even though I found integration tests pretty good tutorial of its own, a basic one might be helpful to encourage the upgrade.

mensfeld commented on lib/karafka/cli/install.rb:
```diff
@@ -49,7 +48,7 @@ def call
           template = File.read(Karafka.core_root.join("templates/#{source}"))
           # @todo Replace with the keyword argument version once we don't have to support
           # Ruby < 2.6
```
I actuyally dont remember with what I was suppose to replace it :laughing:  my guess is, I did not remove the comment...

mensfeld commented:
@opti 
> Great work!
>
> Any plans on adding 1.4 -> 2.0 upgrade guide? Even though I found integration tests pretty good tutorial of its own, a basic one might be helpful to encourage the upgrade.

Yes. That is next on my todo list once I'm done with some more integration tests and readme. The overall plan for the docs is:

- updated readme
- updated changelog
- updated example apps
- upgrade guide
- wikis

Working now on the first two.
